File: pdftract-dotnet/src/Pdftract/Models/Match.cs
using MessagePack;

namespace Pdftract.Models;

/// <summary>
/// Represents a search match result.
/// </summary>
[MessagePackObject]
public record Match
{
    /// <summary>
    /// Page number where the match was found.
    /// </summary>
    [Key(0)]
    public required int PageNumber { get; init; }

    /// <summary>
    /// Matched text content.
    /// </summary>
    [Key(1)]
    public required string Text { get; init; }

    /// <summary>
    /// Surrounding context for the match.
    /// </summary>
    [Key(2)]
    public string? Context { get; init; }
}

File: sdk/php/src/Pdftract/Models/Match.php
<?php

declare(strict_types=1);

namespace Jedarden\Pdftract\Models;

/**
 * Readonly match model
 *
 * Simple readonly representation of a content match within a document
 */
class Match
{
    /**
     * @param int $page Page number where the match was found (1-based)
     * @param string $context Text context surrounding the match
     * @param int $startIndex Starting character index of the match
     * @param int $endIndex Ending character index of the match
     */
    public function __construct(
        public readonly int $page,
        public readonly string $context,
        public readonly int $startIndex,
        public readonly int $endIndex
    ) {}
}
